### PR TITLE
container-linux-config-transpiler: init at 0.4.1

### PR DESCRIPTION
**Pull request by pop**
###### Motivation for this change

Adds a package.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Things to do

I am very close to finishing this, there is just one catch. The output of running `nix-build -A ct.all` produces:

```
$ ls result-bin/bin/
internal  tools
```

The `internal` binary should be called `ct` and the `tools` binary should be discarded. I do not know ho to do this and despite my best efforts I have not been able to figure out the best way to fix this issue. 

Once I am able to rename the binary the package should be ready to be merged, barring any welcomed feedback from the nixpkgs maintainers.

This is my first time contributing to nixpkgs. I am very excited to help the project. Feedback is welcome.

### CODE REVIEW

File: pkgs/development/tools/ct/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  postInstall = ''
+    mv $bin/bin/{internal,ct}
+    rm $bin/bin/tools
+  '';
+
+  meta = {
+    description = "Convert a Container Linux Config into Ignition";
+    license = licenses.asl20;
+    homepage = https://github.com/coreos/container-linux-config-transpiler;
+    maintainers = with maintainers; [offline];
```
Comment (Mic92) [inline pkgs/development/tools/ct/default.nix]: Did @offline agreed to this?

Comment (ElijahCaine) [inline pkgs/development/tools/ct/default.nix]: Oops, that comes from copy-pasting some old stuff. Thanks for the commit @Mic92, I'll add myself to the maintainers.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -535,6 +535,8 @@ with pkgs;
 
   bonnie = callPackage ../tools/filesystems/bonnie { };
 
+  ct = callPackage ../development/tools/ct { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Maybe this should be renamed to `container-linux-config-transpiler` to match the project name. The binary should still named `ct`.

File: pkgs/development/tools/ct/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  postInstall = ''
+    mv $bin/bin/{internal,ct}
+    rm $bin/bin/tools
+  '';
+
+  meta = {
+    description = "Convert a Container Linux Config into Ignition";
+    license = licenses.asl20;
+    homepage = https://github.com/coreos/container-linux-config-transpiler;
+    maintainers = with maintainers; [ElijahCaine];
```
Comment (Mic92) [inline pkgs/development/tools/ct/default.nix]: you need to add yourself to `lib/maintainers.nix` as well.

**Comment by ElijahCaine** [comment]
@Mic92 Thank you for the feedback. Let me know if there's anything else I should do.

**Pull request merged**